File: solutions/golang/loggingframework/console_appender.go
package loggingframework

import "fmt"

type ConsoleAppender struct{}

func NewConsoleAppender() *ConsoleAppender {
	return &ConsoleAppender{}
}

func (a *ConsoleAppender) Append(message *LogMessage) error {
	fmt.Println(message.String())
	return nil
}

File: solutions/java/src/loggingframework/appender/ConsoleAppender.java
package loggingframework.appender;

import loggingframework.formatter.LogFormatter;
import loggingframework.LogMessage;

public class ConsoleAppender implements LogAppender {
    private final LogFormatter formatter;

    public ConsoleAppender(LogFormatter formatter) {
        this.formatter = formatter;
    }

    @Override
    public void append(LogMessage logMessage) {
        System.out.println(formatter.format(logMessage));
    }
}